x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/fielddata/ShapeValues.java

```diff
@@ -113,6 +118,14 @@ public void reset(BytesRef bytesRef) throws IOException {
             this.boundingBox.reset(reader.getExtent(), encoder);
         }
 
+        protected void reset(StreamInput in) throws IOException {
+            int length = in.readInt();
```

Comment:
Suggest `in.readArraySize()` to match the `out.writeVInt()`, plus some extra checks to make sure we don't allocate GiBs of memory just because the stream was corrupted underneath us.

Reply:
`readArraySize()` was protected, but `readByteArray()` calls it and does the checks you suggest, so I'll switch to that.